### ISSUE TITLE
why intl always resolve failed.

### ISSUE THREAD

**Issue by parcool** [singerdmx/flutter-quill#2409]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/singerdmx/flutter-quill/issues)

### The question

Why doesn't flutter_quill update its intl package dependency? It causes version conflicts. I noticed flutter_quill is using an intl package version from a year ago, even though flutter_quill itself maintains very active updates.

```log
/Users/xxx/Flutter/SDK/bin/flutter --no-color pub get
Resolving dependencies...
Because flutter_quill 10.8.5 depends on intl ^0.19.0 and no versions of flutter_quill match >10.8.5 <11.0.0, flutter_quill ^10.8.5 requires intl ^0.19.0.
So, because xxx depends on both intl ^0.20.1 and flutter_quill ^10.8.5, version solving failed.


You can try the following suggestion to make the pubspec resolve:
* Consider downgrading your constraint on intl: flutter pub add intl:^0.19.0
Process finished with exit code 1
```
<img width="962" alt="image" src="https://github.com/user-attachments/assets/15c45c9b-64af-4d68-8171-e3c23e0912be">

**Comment by parcool** [comment]
I apologize, my memory is a bit fuzzy.

**Comment by EchoEllet** [comment]
> Why doesn't flutter_quill update its intl package dependency? 

The versions in `pubspec.yaml` contain the minimum required versions, not the versions that will be used in your project (`pubspec.lock`).

> It causes version conflicts

Requiring newer versions will cause conflicts, too, especially with projects that use an older version of Dart.

The version `0.20.0` is not breaking. We can update the version constraint if needed.

Could you explain what the issue was?

**Comment by EchoEllet** [comment]
Fixed by #2416
